### ISSUE TITLE
Confusing error message when a speed hasn't been given

### ISSUE THREAD

**Issue by ixjlyons** [moorepants/resonance#137]
I've seen this a handful of times now. Here's the traceback you get when you construct a `SingleDoFLinearSystem` and give a coordinate but not a speed:

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-7-fd8087bf5ba0> in <module>()
----> 1 sys.free_response(1)

~/src/resonance/resonance/system.py in free_response(self, final_time, initial_time, sample_rate, **kwargs)
    553         times = self._calc_times(final_time, initial_time, sample_rate)
    554 
--> 555         pos, vel, acc = self._generate_state_trajectories(times, **kwargs)
    556 
    557         self.result = self._state_traj_to_dataframe(times, pos, vel, acc)

~/src/resonance/resonance/linear_systems.py in _generate_state_trajectories(self, times)
    336         sol_func = self._solution_func()
    337 
--> 338         return sol_func(times)
    339 
    340     def period(self):

~/src/resonance/resonance/linear_systems.py in _no_damping_solution(self, time)
    202 
    203         x0 = list(self.coordinates.values())[0]
--> 204         v0 = list(self.speeds.values())[0]
    205 
    206         c1 = v0 / wn

IndexError: list index out of range
```

Perhaps catching this and giving a more direct error message would be helpful.

**Comment by ixjlyons** [comment]
This may be more annoying to catch than I thought since there are quite a few different places this would need to be checked (free response, forced response, linear/nonlinear, etc.). I'm not seeing a single location to check, but maybe it makes sense to have all of the methods that actually do any simulation run a "check system" function at the top.

Maybe kill two birds here and automatically add a speed for each coordinate -- I know that's marked as a TODO in the code.

**Comment by moorepants** [comment]
A check system method sounds good. Probably several things could move in there. Not sure I want to automatically create speeds anymore. That was the original idea, but I moved away from it. If you don't make the speeds explicit it becomes odd discussing setting initial conditions for them.

**Comment by ixjlyons** [comment]
Closed by #144
